### PR TITLE
GL: set texture bias

### PR DESCRIPTION
**Pull request by raven02**

### CODE REVIEW

File: rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ namespace
 		case CELL_GCM_TEXTURE_A1R5G5B5:
 		case CELL_GCM_TEXTURE_R5G6B5:
 		case CELL_GCM_TEXTURE_A8R8G8B8:
-		case CELL_GCM_TEXTURE_R6G5B5:
+		case CELL_GCM_TEXTURE_R6G5B5: return { GL_ALPHA, GL_GREEN, GL_RED, GL_BLUE };
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp]: This one worries me. Can someone check if terraria is affected? Coz i remember tinkering with this when I first started looking at rpcs3 source and setting all those formats to AGRB will interfere. Or maybe you intended to alter R6G5B5 only?

File: rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp [hunk 2]
```diff
@@ -463,9 +463,13 @@ namespace rsx
 
 			glTexParameteri(m_target, GL_TEXTURE_COMPARE_FUNC, gl_tex_zfunc[tex.zfunc()]);
 
-			// TODO: Handle texture bias
-			glTexParameteri(m_target, GL_TEXTURE_MIN_LOD, (tex.min_lod() >> 8));
-			glTexParameteri(m_target, GL_TEXTURE_MAX_LOD, (tex.max_lod() >> 8));
+			glTexEnvi(GL_TEXTURE_FILTER_CONTROL, GL_TEXTURE_LOD_BIAS, (GLint)tex.bias());
+			
+			if (m_target != GL_TEXTURE_RECTANGLE)
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp]: I think native TEXTURE_RECTANGLE support was dropped in favor of emulated unnormalized access in the shaders. m_target can only be 1D, 2D, 3D or CUBE now if i'm not mistaken. I remember having a conversation with vlj about this.

File: rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp (hunk 1 above)
Comment (raven02) [inline rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp]: Let me check with Terraria to see if it is affected .

Comment (raven02) [inline rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp]: I remember it works okay in 0.0.0.6 but looks like it got access violation .Probably blackout this change this moment.

{PPU[0x37] Thread (EE RenderLoop) [0x004c7a44]} struct vm::access_violation thrown: Access violation reading address 0x10

Comment (kd-11) [inline rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp]: @raven02 I'll try to see if I can get terraria to load with some workarounds. It uses the 16-wide formats extensively in the menus. Which other games affected by this change?

Comment (raven02) [inline rpcs3/Emu/RSX/GL/rsx_gl_texture.cpp]: @kd-11 , should be okay now , i miss out the cellsre.sprx.elf

**Comment by kd-11** [comment]
Should be ok as long as tex.bias() returns sane values (some texture control parameters require bitwise manipulation). Also worth noting is that lod bias is often given as a float since even small values have visible effects.

**Comment by kd-11** [comment]
@raven02  Actually, part of my previous comment was valid (about the use of float). use gltexenvf and leave the parameter as a float. lod values will often be fractions.

**Comment by raven02** [comment]
Thanks @kd-11 

@Nekotekina Should be okay to merge

**Comment by danilaml** [comment]
@kd-11 should it be `glTexEnvi` or `gltexenvf`?

**Comment by kd-11** [comment]
texenvf

**Comment by kd-11** [comment]
@raven02 the method to call for float upload should be texenvf, with the last value left as a float:

i.e
glTexEnvf(GL_TEXTURE_FILTER_CONTROL, GL_TEXTURE_LOD_BIAS, tex.bias());

**Comment by raven02** [comment]
Let me quick fix it with a pull

**Pull request merged**